crates/ruff_formatter/src/format_element.rs

```diff
@@ -403,6 +399,47 @@ pub trait FormatElements {
     fn end_tag(&self, kind: TagKind) -> Option<&Tag>;
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum TextWidth {
+    Width(NonZeroU32),
+    Multiline,
+}
+
+impl TextWidth {
+    pub(crate) const fn new_width(width: u32) -> Self {
+        // SAFETY: 1 + x is guaranteed to be non zero
+        #[allow(unsafe_code)]
+        TextWidth::Width(unsafe { NonZeroU32::new_unchecked(width.saturating_add(1)) })
```

Comment:
What about `NonZeroU32::MIN.saturating_add(width)` or `NonZerou32::new(1).unwrap().saturating_add(width)`?

Reply:
please add a comment about the +1/-1 trickery in general